auth_backends/pipeline.py

```diff
@@ -32,10 +34,36 @@ def get_user_if_exists(strategy, details, user=None, *args, **kwargs):  # pylint
 
 
 def update_email(strategy, details, user=None, *args, **kwargs):  # pylint: disable=keyword-arg-before-vararg
-    """Update the user's email address using data from provider."""
+    """Update the user's email address using data from provider.
+    
+    Before updating the email, verify that the username in details matches the user's username.
+    If usernames don't match, log a warning and don't update the email.
+    Custom attributes are set to track username mismatches.
```

Comment:
I appreciate good comment, but maybe these details could just be left to well commented code below. This is a judgement call. What you have is not wrong. I'd personally go lighter here. The more text, the more comments can drift from code.